lib/bots/versionbot.ts

```diff
@@ -212,11 +212,38 @@ export class VersionBot extends GithubBot.GithubBot {
             // Go through all the commits. We're looking for, at a minimum, a 'change-type:' tag.
             for (let commit of commits) {
                 const commitMessage: string = commit.commit.message;
-                const invalidCommit = !commitMessage.match(/^change-type:\s*(patch|minor|major)\s*$/mi);
 
-                if (!invalidCommit) {
-                    changetypeFound = true;
-                    break;
+                // Split the commits up into lines, and find the last line with any whitespace in.
+                // Whilst we tend to ask for:
+                //  <header>
+                //
+                //  <body>
+                //
+                //  <footer>
+                // This code will actually let you get away with:
+                //  <header>
+                //
+                //  <footer>
+                // As sometimes a development patch may be self-explanatory in the header alone.
+                const lines = commitMessage.split('\n');
+                let index = 0;
+                for (index = lines.length - 1; index >= 0; index -= 1) {
+                    if (lines[index].match(/$\s*^/)) {
+                        break;
+                    }
+                }
+                // If there's no match, then at the very least there's no footer, and the commit
+                // is in the wrong format (as there's no text to use in the logs).
+                if (index > 0) {
+                    // We should have a line index to join from, now.
+                    lines.splice(0, index);
+                    const footer = lines.join('\n');
+                    const invalidCommit = !footer.match(/^change-type:\s*(patch|minor|major)\s*$/mi);
+
+                    if (!invalidCommit) {
+                        changetypeFound = true;
+                        break;
+                    }
```

Comment:
If we push through https://github.com/resin-io/versionist/issues/62 it would mean that we could use that to check that the commit was valid, keeping versionist and VersionBot tightly coupled.